cpp/Platform.Equality/Platform.Equality.Any.h

```diff
@@ -0,0 +1,101 @@
+#pragma once
+
+#ifndef PLATFORM_EQUALITY_ANY_H
+#define PLATFORM_EQUALITY_ANY_H
+
+#include <any>
+#include <functional>
+#include <typeindex>
+#include <concepts>
+#include <stdexcept>
+
+namespace Platform::Equality
+{
+    template<class T>
+    inline auto __to_any_equal_visitor(auto func)
+    {
+        return std::pair<std::type_index, std::function<bool(std::any, std::any)>>
+        {
+                std::type_index(typeid(T)),
+                [func](std::any a, std::any b) -> std::size_t
```

Comment:
```cpp
[func = std::move(func)](const std::any& a, const std::any& b)
```

Reply:
Wouldn't [func] cause a lambda-capture?

Reply:
`[func]` will make a copy of the functor, ```[func = std::move(funс)]``` will move the object